### ISSUE TITLE
controllers for functionality 

### ISSUE THREAD

**Issue by ameet20** [scalatra/scalatra#874]
In particular, use of multiple ScalatraFilters makes deep stack. It would be nice if we can composite them to a single controller and they are proceeded by loop rather than nested filters chain

**Comment by takezoe** [comment]
Good point! Just from that reason, we use a filter which allows to composite multiple ScalatraFilters in GitBucket which is an open source git server based on Scalatra:
https://github.com/gitbucket/gitbucket/blob/master/src/main/scala/gitbucket/core/servlet/CompositeScalatraFilter.scala

But I'm not sure if this filter should be a part of Scalatra itself.
